File: 1803-average-waiting-time/average-waiting-time.py
class Solution(object):
    def averageWaitingTime(self, customers):
        start = customers[0][0]
        end = start + customers[0][1]
        total_waiting_time = float(end-start)
        for i in range(1, len(customers)):
            if(customers[i][0] <= end):
                start = end
                end = start + customers[i][1]
            else:
                start = customers[i][0]
                end = start + customers[i][1]
            total_waiting_time += (end - customers[i][0])

        avg_waiting_time = total_waiting_time/(len(customers))
        return avg_waiting_time
        

File: 1803-average-waiting-time/average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        int start = customers[0][0], end = start + customers[0][1];
        double total_waiting_time = end-start;
        for(int i=1; i<customers.size(); i++){
            if(customers[i][0] <= end){
                start = end;
                cout << start << endl;
                end = start + customers[i][1];
            }
            else{
                start = customers[i][0];
                end = start + customers[i][1];
            }
            total_waiting_time += (end - customers[i][0]);
        }
        double avg_waiting_time = total_waiting_time/(customers.size());
        return avg_waiting_time;
    }
};